codalab/worker/file_util.py

```diff
@@ -531,15 +531,26 @@ def get_path_size(path, exclude_names=[], ignore_nonexistent_path=False):
             return 0
         # Raise the FileNotFoundError
         raise
+
+    # Detect and ignore Stale file handler. Please see: https://www.baeldung.com/linux/stale-file-handles
     if not os.path.islink(path) and os.path.isdir(path):
-        for child in os.listdir(path):
-            if child not in exclude_names:
-                try:
-                    full_child_path = os.path.join(path, child)
-                except UnicodeDecodeError:
-                    full_child_path = os.path.join(path.decode('utf-8'), child.decode('utf-8'))
-                result += get_path_size(full_child_path, ignore_nonexistent_path=True)
-    return result
+        try:
+            child_paths = os.listdir(path)
+        except OSError:
+            if ignore_nonexistent_path:
+                # If we have trouble list the dir, return the size of this path as 0
+                # Do not raise error and just ignore the stale file handler/
+                return 0
```

Comment:
Could this potentially suppress other errors?  Maybe worth printing out a warning at least (assuming it doesn't result in too much output)?

Reply:
Thank you! I will add a Warning print with the path, and test it out using the dev environment before getting it merged.